CONTRIBUTING.md

```diff
@@ -17,63 +19,97 @@ Jdbi places serious emphasis on not breaking compatibility. Remember these simpl
 
 Due to the volume of feedback in a typical PR, we may push changes directly to your PR branch if we are able to, in order to save time and frustration for everyone.
 
-# Enable `-parameters` compiler flag in your IDE:
+# Setup
+
+## Enable `-parameters` compiler flag
 
 Most of our SQL Object tests rely on SQL method parameter names. However by default, `javac` does not compile these
 parameter names into `.class` files. Thus, in order for unit tests to pass, the compiler must be configured to output
 parameter names.
 
-## IntelliJ
+### IntelliJ
 
-* File -> Settings
-* Build, Execution, Deployment -> Compiler -> Java Compiler
+* File &rarr; Settings
+* Build, Execution, Deployment &rarr; Compiler &rarr; Java Compiler
 * Additional command-line parameters: `-parameters`
 * Click Apply, then OK.
-* Build -> Rebuild Project
+* Build &rarr; Rebuild Project
+
+## Maven
+
+### Building individual modules
+
+Maven multi-module projects with cross-module dependencies such as plugin configurations with relative file paths are nowadays still a pain to get right.
+When building jdbi from the parent module, everything will normally work fine, as that's how it is built on our CI.
+When trying to e.g. `mvn clean package` any individual module, you may encounter an error such as this:
+
+> java -Dmaven.multiModuleProjectDirectory=D:/Projects/Java/jdbi/core
+> -Dfoo=bar maven.jar clean package  
+> ...  
+> ...  
+> [INFO] BUILD FAILURE  
+> [INFO] ------------------------------------------------------------------------  
+> [INFO] Total time: 16.767 s  
+> [INFO] Finished at: 2018-12-07T20:55:19+01:00  
+> [INFO] ------------------------------------------------------------------------  
+> [ERROR] Failed to execute goal org.apache.maven.plugins:maven-pmd-plugin:3.9.0:pmd (pmd) on project jdbi3-core:
+> Execution pmd of goal org.apache.maven.plugins:maven-pmd-plugin:3.9.0:pmd failed:
+> org.apache.maven.reporting.MavenReportException:
+> **Could not find resource 'D:/Projects/Java/jdbi/core/src/build/pmd.xml'**. -> [Help 1]
+
+This typically happens when your IDE incorrectly sets the `maven.multiModuleProjectDirectory` property for maven, as in the example above.
+This property _should_ point to the _project root_, where the global files for PMD, CheckStyle, and others are housed.
+In the example, you can see it actually points to the root of _the module that was built_ — `core` in this case — which causes a global file to not be found.
+
+#### Intellij
+
+* File &rarr; Settings
+* Build Tools &rarr; Maven &rarr; Runner
+* VM Options: <code>-Dmaven.multiModuleProjectDirectory=D:/Projects/Java<b>/jdbi</b></code>
 
-# Setting up your machine to build `jdbi3-oracle12`
+## Set up your machine to build `jdbi3-oracle12`
```

Comment:
Ha, this whole section is going away :boot:

Reply:
care to join me on discord if you're online after all?